File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isVowel(char c)
    {
        if (c == 'A' || c == 'E' || c == 'I' || c == 'O' ||
                c == 'U' || c == 'a' || c == 'e' || c == 'i' ||
                c == 'o' || c == 'u')
                {
                    return true;
                }
        return false;
    }
    string reverseVowels(string s) {
        int i=0,j=s.size()-1;
        while(i<j)
        {
            bool a=isVowel(s[i]),b=isVowel(s[j]);
            if( a && b)
            {
                swap(s[i],s[j]);
                i++;j--;
            }
            else if(a)
            {
                j--;
            }
            else{
                i++;
            }


        }
        return s;
    }
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    private boolean isVowel(char c)
    {
        if(c=='A' || c=='E' || c=='I' || c=='O' || c=='U' || 
        c=='a' || c=='e' || c=='i' || c=='o' || c=='u' )
        {
            return true;
        }

        return false;
    }
    public String reverseVowels(String s) {
        char[] ch = s.toCharArray();
        int n = ch.length;
        int j=n-1,i=0;
        while(i<j)
        {
            if(isVowel(ch[i]))
            {
                while(!isVowel(ch[j]))
                {
                    j--;
                }
                char c= ch[i];
                ch[i]=ch[j];
                ch[j]=c;
                j--;
            }
            i++;
        }
        StringBuilder ans = new StringBuilder();
        for(char c : ch)
        {
            ans.append(c);
        }
        return ans.toString();
    }
}